src/cqparts/codec/amf.py

```diff
@@ -30,10 +30,17 @@ def __call__(self, filename='out.amf', world=False, tolerance=0.1):
         shape = workplane.val()
 
         # call cadquery exporter
-        with open(filename, 'wb') as fh:
-            cadquery.freecad_impl.exporters.exportShape(
-                shape=shape,
-                exportType='AMF',
-                fileLike=fh,
-                tolerance=tolerance,
-            )
+        try:
```

Comment:
remove freecad/occ switch

Also, this implementation will open a file handle for writing before it encounters the exception, so it will create the handle twice.